### PR TITLE
Re-implement InitCacheFlushTimeout option and tests

### LINKED ISSUE #1642: Remove BlockCacheFlush and improve InitCacheFlushTimeout mechanism
The option `InitCacheFlushTimeout` controls how the [`BlockCacheFlush`](https://github.com/getsentry/sentry-dotnet/blob/fc5f464ff0245676613764cf7e45597e151dc6c3/src/Sentry/Internal/SdkComposer.cs#L55) method works.  Unfortunately, there are issues with this method, such as those raised in #1640.  Mostly this is caused by sync-over-async antipattern.

I was going to rewrite most of it to be async, but after reviewing carefully, I'm fairly certain we don't need this feature at all.  I'd like to remove it.

The only thing it does is pauses initialization on the calling thread for a certain amount of time while it tries to flush the cache in a background task.  Either the cache is flushed within the timeout specified, or the timeout is reached and the flush keeps occurring in the background.

The reason I don't think we need it is that the `CachingTransport` *already* works in the background.  As soon as it is created, it starts a background task and enters a loop, waiting for a signal that something is being sent.  Then it processes items (same as flushing) before reseting the signal.

I propose we simply invert the loop.  Always start with processing/flushing items, then wait for the signal that there's more items being sent.  That will have the same effect as `BlockFlushCache` was doing, but without any need to worry about the `InitCacheFlushTimeout` and without any delay in completing initialization.

Associated PR forthcoming.

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
Fixes #1642 and #1640

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## Unreleased
+
+### Fixes
+- Rework how the `InitCacheFlushTimeout` option is implemented. ([#1644](https://github.com/getsentry/sentry-dotnet/pull/1644))
```
Comment (bruno-garcia) [inline CHANGELOG.md:6]:
```suggestion

- Rework how the `InitCacheFlushTimeout` option is implemented. ([#1644](https://github.com/getsentry/sentry-dotnet/pull/1644))
```

File: src/Sentry/Internal/Http/CachingTransport.cs [hunk 2]
```diff
@@ -80,6 +82,20 @@ private void Initialize(bool startWorker)
...
             _worker = startWorker ? Task.Run(CachedTransportBackgroundTaskAsync) : Task.CompletedTask;
+
+            // Wait for init timeout, if configured.  (Can't do this without a worker.)
+            if (startWorker && _options.InitCacheFlushTimeout > TimeSpan.Zero)
+            {
+                using (_initCacheResetEvent = new ManualResetEventSlim())
+                {
+                    // This will complete either when the first round of processing is done,
+                    // or on timeout, whichever comes first.
+                    _initCacheResetEvent.Wait(_options.InitCacheFlushTimeout);
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Http/CachingTransport.cs:93]: Is it worth adding this within a `if` and logging the outcome?

**Review by bruno-garcia — APPROVED** [review]
Much simpler solution, thanks Matt!

File: src/Sentry/Internal/Http/CachingTransport.cs (hunk 2 above)
Comment (mattjohnsonpint) [reply src/Sentry/Internal/Http/CachingTransport.cs:93]: Probably.  Will do. Thanks.

**Pull request merged**